cmd/starter/c/starter.c

```diff
@@ -214,12 +214,13 @@ static struct capabilities *get_process_capabilities() {
 
 static int get_last_cap(void) {
     int last_cap;
-    for ( last_cap = CAPSET_MAX; ; last_cap-- ) {
-        if ( prctl(PR_CAPBSET_READ, last_cap) > 0 || last_cap == 0 ) {
+    for ( last_cap = CAPSET_MIN; last_cap <= CAPSET_MAX; last_cap++ ) {
+        if ( prctl(PR_CAPBSET_READ, last_cap) < 0 ) {
+            /* an error means that capability is not valid, take the last valid */
             break;
         }
     }
-    return last_cap;
+    return --last_cap;
```

Comment:
If `CAPSET_MAX=37` is the true highest cap, then shouldn't we be returning `last_cap` instead of `last_cap--` if we reach it in the loop? Or have I misunderstood? Thanks.

Reply:
When completing the loop `for ( last_cap = CAPSET_MIN; last_cap <= CAPSET_MAX; last_cap++ )` `last_cap` is 38 to match `<=` condition, if an error occurs with `last_cap = 37` during prctl call it means max is 36, so `last_cap` need to be decremented in all cases

Reply:
yes, sorry - silly misread of the loop condition on my part. Apologies!